### PR TITLE
Has/change bool flag to keyword list

### PR DESCRIPTION
**Pull request by hamir-suspect**
@radwo  this is different implementation WDYT

### CODE REVIEW

File: lib/watchman.ex [hunk 1]
```diff
@@ -14,28 +14,49 @@ defmodule Watchman do
     Supervisor.start_link(children, opts)
   end
 
-  def submit(name, value, type \\ :gauge) do
-    Watchman.Server.submit(name, value, false, type)
+  def submit(name, value, type \\ :gauge)
+
+  def submit({external, name}, value, type) when external in [:internal, :external, :always] do
```
Comment (radwo) [inline lib/watchman.ex]: We shouldn't name this variable `external` anymore.

File: lib/watchman.ex [hunk 2]
```diff
@@ -14,28 +14,49 @@ defmodule Watchman do
...
+    Watchman.Server.submit([{external, name}], value, type)
+  end
+
+  def submit(name_list, value, type) when is_list(name_list) do
```
Comment (radwo) [inline lib/watchman.ex]: `name_list` -> `names` - to make it simple

File: lib/watchman.ex [hunk 3]
```diff
@@ -14,28 +14,49 @@ defmodule Watchman do
...
+    Watchman.Server.submit([{:internal, name}], value, type)
+  end
+
+  def increment(name_list) when is_list(name_list) do
+    submit(name_list, 1, :count)
   end
 
-  def submit(name, value, external, type) do
-    Watchman.Server.submit(name, value, external, type)
+  def increment({external, name}) when external in [:internal, :external, :always] do
```
Comment (radwo) [inline lib/watchman.ex]: `increment`, `decrement`, and `benchmark` are using `submit`, so maybe we do not need to duplicate the `name` logic in them?

**Review by radwo — COMMENTED** [review]
Check also this comment from Damjan https://github.com/renderedtext/ex-watchman/pull/26#issuecomment-1302314052

**Comment by radwo** [comment]
@DamjanBecirovic, can you take a look at this alternative approach? In which, we will be able to use one Watchman call. It will be handier, especially in `benchmark` calls. 

It should work like that:

To send only internal metrics and  keep the back compatibility:
```
Watchman.submit("some.internal.metric.name", count, :gauge)
``` 

To send only internal or external metrics with the specific name
```
Watchman.submit({:internal, "some.internal.metric.name"}, count, :gauge) 
```

To send both metrics with different names
```
Watchman.submit([{:internal, "some.internal.metric.name"}, {:external, "some.external.metric.name"}], count, :gauge) 
```

**Comment by DamjanBecirovic** [comment]
The approach looks good, mainly since it effectively supports both two Watchman calls and one call with two names provided.
The one thing I'm not sure about is how this works with tags and whether it will support different tags for internal/external metrics.
Can we add a test for that?

**Comment by hamir-suspect** [comment]
> The approach looks good, mainly since it effectively supports both two Watchman calls and one call with two names provided. The one thing I'm not sure about is how this works with tags and whether it will support different tags for internal/external metrics. Can we add a test for that?

here is one way to put it, and another way is to use keyword list, so it can either be
``` Watchman.increment([external: {"name", ["tag"]}])```
or
``` Watchman.increment({:external, {"name", ["tag"]}})```
https://github.com/renderedtext/ex-watchman/pull/27/files#diff-72b38da73bfe7168355fc7b7091644e7576761488f855350229d643ab9d050faR246

**Review by DamjanBecirovic — APPROVED** [review]

**Pull request merged**